terraform/base/route53.tf

```diff
@@ -16,217 +32,220 @@ resource "aws_route53_record" "heroku-coalease-cname" {
     ttl = "180"
     records = ["oita-54541.herokussl.com"]
 }
-resource "aws_route53_record" "heroku-archiver-cname-prod" {
+
+
+resource "aws_route53_record" "heroku-archiver-cname-stage" {
     zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "archiver.mozilla-releng.net"
+    name = "archiver.staging.mozilla-releng.net"
     type = "CNAME"
     ttl = "180"
-    records = ["archiver.mozilla-releng.net.herokudns.com"]
+    records = ["archiver.staging.mozilla-releng.net.herokudns.com"]
 }
-resource "aws_route53_record" "heroku-clobberer-cname-prod" {
+
+
+resource "aws_route53_record" "heroku-clobberer-cname-stage" {
     zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "clobberer.mozilla-releng.net"
+    name = "clobberer.staging.mozilla-releng.net"
     type = "CNAME"
     ttl = "180"
     records = ["saitama-70467.herokussl.com"]
 }
-resource "aws_route53_record" "heroku-mapper-cname-prod" {
-    zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "mapper.mozilla-releng.net"
-    type = "CNAME"
-    ttl = "180"
-    records = ["mapper.mozilla-releng.net.herokudns.com"]
-}
-resource "aws_route53_record" "heroku-tooltool-cname-prod" {
+
+
+resource "aws_route53_record" "heroku-docs-cname-prod" {
     zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "tooltool.mozilla-releng.net"
-    type = "CNAME"
-    ttl = "180"
-    records = ["kochi-11433.herokussl.com"]
+    name = "docs.mozilla-releng.net"
+    type = "A"
+    alias {
+        name = "d1945er7u4liht.cloudfront.net."
+        zone_id = "Z2FDTNDATAQYW2"
+        evaluate_target_health = false
+    }
 }
-resource "aws_route53_record" "heroku-treestatus-cname-prod" {
+
+
+resource "aws_route53_record" "heroku-docs-cname-stage" {
     zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "treestatus.mozilla-releng.net"
-    type = "CNAME"
-    ttl = "180"
-    records = ["treestatus.mozilla-releng.net.herokudns.com"]
+    name = "docs.staging.mozilla-releng.net"
+    type = "A"
+    alias {
+        name = "d32jt14rospqzr.cloudfront.net."
+        zone_id = "Z2FDTNDATAQYW2"
+        evaluate_target_health = false
+    }
 }
 
 
-###############################
-## Heroku staging app cnames ##
-###############################
-
-resource "aws_route53_record" "heroku-archiver-cname-stage" {
+resource "aws_route53_record" "heroku-frontend-cname-prod" {
     zone_id = "${aws_route53_zone.mozilla-releng.zone_id}"
-    name = "archiver.staging.mozilla-releng.net"
-    type = "CNAME"
-    ttl = "180"
-    records = ["archiver.staging.mozilla-releng.net.herokudns.com"]
+    name = "mozilla-releng.net"
```

Comment:
This record conflicts with "aws_route53_record" "root-alias".  This should be name = "www.mozilla-releng.net" which is what the root alias points at.  Also, this resource record is named 'heroku-frontend-cname-prod',  it is neither a heroku app nor a cname record.  We either need to stay consistent and use the resource name to indicate the type of record (cname vs A record alias) or we just drop that and make each name equivalent to the record name with dot and dash converted to underscore.  This might make it easier to read and to code.  Keep in mind that changing the resource name will cause the records to destroyed and recreated when ran through terraform.  It will treat which one as a completely new resource.  This might lead to a brief interruption of dns service for the records affected.

eg. "aws_route53_record" "cloudfront-frontend-alias-prod" or "aws_route53_record" "mozilla_releng_net"